go/arrow/array/list.go

```diff
@@ -52,6 +52,13 @@ func NewListData(data arrow.ArrayData) *List {
 
 func (a *List) ListValues() arrow.Array { return a.values }
 
+func (a *List) ValueStr(i int) string {
+	if !a.IsValid(i) {
+		return NullValueStr
+	}
+	return fmt.Sprintf("%v", a.newListValue(i))
```

Comment:
same comment as for FixedSizeList. needs to defer call release on the return from `newListValue` and please add a test that verifies no leaked memory